glos_qartod/run.py

```diff
@@ -57,41 +57,46 @@ def qc_subset(dir_root, conf, mappings):
 
 
 def find_files(dest_dir, qc_varnames, qc_varnames_bkp):
-    files = []
+    nc_files = []
     if os.path.exists(dest_dir):
         for root, subdir, fname in os.walk(dest_dir):
-            for nc_file in (f for f in fname if f.endswith('.nc')):
-                file_dest = os.path.join(root, nc_file)
-                # check if all the variables already exist in the file
-                # if all of the variables for the expected QC variables
-                # aren't present in the dataset, add the file to the list
-                # of files to be QCed
+            # find all .nc files in this directory, check if there has been
+            # qc applied to them, and create absolute paths to them
+            # path to them
+
+            # higher order function to use in checking listcomp
+            def pred_fn(filepath):
+                return check_if_qc_vars_exist(os.path.join(root, filepath),
+                                              qc_varnames, qc_varnames_bkp)
```

Comment:
Do not dynamically create functions in the middle of a function. There's very few good uses of it, especially in a loop, and this is not one of them.